scripts/download.pl

```diff
@@ -70,20 +70,41 @@ ()
 	return undef;
 }
 
-sub download_cmd($) {
+sub download_cmd {
 	my $url = shift;
 	my $have_curl = 0;
+	my $have_aria2c = 0;
+	my $fn = shift;
+	my @mrs = @_;
+	my $murl = "'$url'";
 
+	my @chArray = ('a'..'z', 'A'..'Z', 0..9);
+	my $rfn = join '', map{ $chArray[int rand @chArray] } 0..9;
 	if (open CURL, '-|', 'curl', '--version') {
 		if (defined(my $line = readline CURL)) {
 			$have_curl = 1 if $line =~ /^curl /;
 		}
 		close CURL;
 	}
+	if (open ARIA2C, '-|', 'aria2c', '--version') {
+		if (defined(my $line = readline ARIA2C)) {
+			$have_aria2c = 1 if $line =~ /^aria2 /;
+		}
+		close ARIA2C;
+	}
+
+	print ("HAVE_ARIA2C:");
+	print ("$have_aria2c\n");
+
+	for my $el (@mrs) {
+		$murl = $murl." '$el/$fn'";
+	}
 
-	return $have_curl
-		? (qw(curl -f --connect-timeout 20 --retry 5 --location), $check_certificate ? '' : '--insecure', shellwords($ENV{CURL_OPTIONS} || ''), $url)
-		: (qw(wget --tries=5 --timeout=20 --output-document=-), $check_certificate ? '' : '--no-check-certificate', shellwords($ENV{WGET_OPTIONS} || ''), $url)
+	return $have_aria2c
```

Comment:
Think here we should use a full `if() else` than `$have_aria2c ? :` just to make things more readable.

also AFAIK also aria2c supports disabling/enabling TLS verification. Can you add this here for consistency with what curl does?

Reply:
sure i'll do that